tests/testsuite/install.rs

```diff
@@ -2734,3 +2734,132 @@ fn uninstall_running_binary() {
 
 "#]]).run();
 }
+
+#[cargo_test]
+fn dry_run() {
+    pkg("foo", "0.0.1");
+
+    cargo_process("install foo")
+        .with_stderr_data(str![[r#"
+[UPDATING] `dummy-registry` index
+[DOWNLOADING] crates ...
+[DOWNLOADED] foo v0.0.1 (registry `dummy-registry`)
+[INSTALLING] foo v0.0.1
+[COMPILING] foo v0.0.1
+[FINISHED] `release` profile [optimized] target(s) in [ELAPSED]s
+[INSTALLING] [ROOT]/home/.cargo/bin/foo[EXE]
+[INSTALLED] package `foo v0.0.1` (executable `foo[EXE]`)
+[WARNING] be sure to add `[ROOT]/home/.cargo/bin` to your PATH to be able to run the installed binaries
+
+"#]])
+        .run();
+    assert_has_installed_exe(paths::cargo_home(), "foo");
+}
+
+#[cargo_test]
+fn dry_run_incompatible_package() {
+    Package::new("some-package-from-the-distant-future", "0.0.1")
+        .rust_version("1.2345.0")
+        .file("src/lib.rs", "")
+        .file(
+            "src/main.rs",
+            &format!(
+                "extern crate {}; fn main() {{}}",
+                "some-package-from-the-distant-future"
+            ),
+        )
+        .publish();
+
+    cargo_process("install some-package-from-the-distant-future")
+        .with_status(101)
+        .with_stderr_data(str![[r#"
+[UPDATING] `dummy-registry` index
+[ERROR] cannot install package `some-package-from-the-distant-future 0.0.1`, it requires rustc 1.2345.0 or newer, while the currently active rustc version is [..]
```

Comment:
This tests for the error when installing `some-package-from-the-distant-future`.  We also need a test for when `some-package-from-the-distant-future` is a dependency.  That is what will hit `create_bcx` and that section of MSRV code, ensuring we are doing enough of the compile process to hit it.

Reply:
I added a test for when `some-package-from-the-distant-future` is a dependency.